internal/fwschema/diagnostics.go

```diff
@@ -64,3 +64,18 @@ func AttributeDefaultTypeMismatchDiag(attributePath path.Path, expectedType attr
 			"The default value must match the type of the schema.",
 	)
 }
+
+// AttributeInvalidElementTypeDiag returns an error diagnostic to provider
+// developers about using non-primitive types in their Attribute
+// implementation. This is not allowed.
+func AttributeInvalidElementTypeDiag(attributePath path.Path) diag.Diagnostic {
+	// The diagnostic path is intentionally omitted as it is invalid in this
+	// context. Diagnostic paths are intended to be mapped to actual data,
+	// while this path information must be synthesized.
+	return diag.NewErrorDiagnostic(
+		"Invalid Attribute Implementation",
+		"When validating the schema, an implementation issue was found. "+
+			"This is always an issue with the provider and should be reported to the provider developers.\n\n"+
+			fmt.Sprintf("%q contains an Attribute type that is not allowed for Resource Identity. ", attributePath),
+	)
```

Comment:
I think we should consider printing more information in this diagnostic and maybe be a little more specific about what's wrong (since the diagnostic is only used by one validation)

Some wording that mentions that the only valid `list` for resource identity is one that has a primitive element type (bool, int32, etc.), then maybe print out the `attr.Type` that we received?